lsp-bridge.el

```diff
@@ -749,31 +753,33 @@ So we build this macro to restore postion after code format."
             (lsp-bridge-enable-org-babel
              ;; get lang server according to org babel
              (let* ((lang (org-element-property :language lsp-bridge--org-babel-info-cache))
-                    (mode-name (concat (symbol-name (cdr (assoc lang org-src-lang-modes))) "-mode"))
+                    (lang-name (symbol-name (cdr (assoc lang org-src-lang-modes))))
+                    (mode-name (concat lang-name "-mode"))
                     (major-mode (intern mode-name)))
-               (if (eq major-mode 'emacs-lisp-mode)
-                   (setq-local acm-is-elisp-mode-in-org t))
-               (lsp-bridge-get-single-lang-server-by-mode))))))))
+               (setq-local acm-is-elisp-mode-in-org (eq major-mode 'emacs-lisp-mode))
+               ;; if `lsp-bridge-org-babel-lang-list' is set, only enable lsp when lang in it
+               (when (or (eq lsp-bridge-org-babel-lang-list nil)
+                         (member lang-name lsp-bridge-org-babel-lang-list))
+                 (lsp-bridge-get-single-lang-server-by-mode)))))))))
 
 (defvar-local lsp-bridge--org-update-file-before-change nil)
 (defun lsp-bridge-check-org-babel-lsp-server ()
   "Check if current point is in org babel block. "
   (if (and lsp-bridge--org-babel-block-bop lsp-bridge--org-babel-block-eop lsp-bridge--org-babel-info-cache
-           (> (point) lsp-bridge--org-babel-block-bop) (< (point) lsp-bridge--org-babel-block-eop))
+           (>= (point) lsp-bridge--org-babel-block-bop) (<= (point) lsp-bridge--org-babel-block-eop))
       lsp-bridge--org-babel-info-cache
     (setq-local lsp-bridge--org-babel-info-cache (org-element-context))
-    (unless (eq (org-element-type lsp-bridge--org-babel-info-cache) 'src-block)
-      (setq-local lsp-bridge--org-babel-info-cache nil))
-    (when lsp-bridge--org-babel-info-cache
-      (setq-local lsp-bridge--org-babel-block-bop (org-element-property :begin lsp-bridge--org-babel-info-cache))
-      (setq-local lsp-bridge--org-babel-block-eop (org-element-property :end lsp-bridge--org-babel-info-cache))
+    (if (not (eq (org-element-type lsp-bridge--org-babel-info-cache) 'src-block))
+        (setq-local lsp-bridge--org-babel-info-cache nil)
+      (save-excursion
+        (goto-char (org-element-property :begin lsp-bridge--org-babel-info-cache))
```

Comment:
This will fail when src block contains affiliated keywords (like #+name:). Better use :post-affiliated propery.

Reply:
Thanks. I have opened a PR https://github.com/manateelazycat/lsp-bridge/pull/577#issue-1640965705